pkg/extensions/extension_search.go

```diff
@@ -71,12 +71,18 @@ func EnableSearchExtension(config *config.Config, storeController storage.StoreC
 }
 
 func downloadTrivyDB(interval time.Duration, sch *scheduler.Scheduler, cveInfo CveInfo, log log.Logger) {
-	generator := &trivyTaskGenerator{interval, cveInfo, log, pending, 0, time.Now(), &sync.Mutex{}}
+	generator := &TrivyTaskGenerator{interval, cveInfo, log, pending, 0, time.Now(), &sync.Mutex{}}
 
 	sch.SubmitGenerator(generator, interval, scheduler.HighPriority)
 }
 
-type trivyTaskGenerator struct {
+func CreateTrivyTaskGenerator(interval time.Duration, cveInfo CveInfo, log log.Logger) *TrivyTaskGenerator {
```

Comment:
```suggestion
func NewTrivyTaskGenerator(interval time.Duration, cveInfo CveInfo, log log.Logger) *TrivyTaskGenerator {
```
And also call it at line 74.